Add documentation for Network Policies and a Cilium FAQ page (#10)
appuio/managed-openshift-docs: docs/modules/ROOT/pages/cilium/network-policies.adoc

> = Cilium Network Policies

Reviewer: I would create two pages. One just `Network Policies` with `Default Network Policies` subchapter and a warning box linking to a new page "Cilium Cluster Mesh" where all cluster mesh Q&A is written down. Also so we can link it if we ever do more marketing for the product cluster mesh.

> Cilium provides standard network policies for you cluster via the NetworkPolicy resource as well as extended functions via CiliumNetworkPolicy as CustomRessourceDefinition.

Reviewer: ```suggestion
As a Kubernetes Network Plugin plugin Cilium integrates transparently with standard Kubernetes https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.32/#networkpolicy-v1-networking-k8s-io[`NetworkPolicies`].
Cilium also provides extended configuration though https://docs.cilium.io/en/latest/network/kubernetes/policy/#ciliumnetworkpolicy[`CiliumNetworkPolicies`].
Extended policies may allow you to configure L3, L4 and L7 network policies.
Those extended policies are not usually needed and should be avoided if not necessary as they might introduce performance or stability issues.
```

> You can read about network policies here via https://docs.openshift.com/container-platform/4.16/networking/network_security/network_policy/about-network-policy.html[upstream documentation].

Reviewer: ```suggestion
You can read about network policies here via https://kubernetes.io/docs/concepts/services-networking/network-policies/[upstream documentation].
```

IMO much clearer and will not break every time redhat decides to reorganize their shit.

> You can find examples on how to use the extended funcitonality of cilium network policies https://docs.cilium.io/en/latest/network/kubernetes/policy/#ciliumnetworkpolicy[here].

Reviewer: ```suggestion
You can find examples on how to use the extended functionality of cilium network policies https://docs.cilium.io/en/latest/network/kubernetes/policy/#ciliumnetworkpolicy[here].
```

> == Default Network Policiecs

Reviewer: ```suggestion
== Default Network Policies
```

> Most clusters have two default network policies applied. You can list them in your namespace using:

Reviewer: ```suggestion
VSHN Managed OpenShift isolates namespaces by default. It does this by auto-generating two network policies.
You can list them in your namespace using:
```

[source,shell]
--
oc get networkpolicy
--

> 

Reviewer: ```suggestion

[NOTICE]
----
Network Policies are deny by default and additive.
A pod matched by an empty policy does not allow any traffic. 
Every network policy adds to the allowlist.
----
```

> `allow-from-other-namespaces`:: Allows incoming traffic from the monitoring stack, cluster ingresses, and the host network.

Reviewer: ```suggestion
`allow-from-other-namespaces`:: Allows incoming traffic from the monitoring stack, cluster ingresses, and the host network.
Because policies are deny by default it denies all other ingress traffic.
```

`allow-from-same-namespace`:: Allows unrestricted traffic within the same namespace.

== Interaction of Network Policies and Cilium Cluster Mesh

> When using Cilium Cluster Mesh, it's important to document how network policies interact across clusters, as the behavior may be unexpected.

Reviewer: ```suggestion
[WARNING]
----
If using Cilium Cluster Mesh some identifies thought as unique in a cluster might no longer be unique.
----
```

> The default allow-from-same-namespace policy applies within an OpenShift namespace. While namespaces are unique within a single cluster, this constraint does not extend to a meshed cluster. Since the policy matches based on namespace name, it applies to any namespace with the same name across clusters. For example, if you have a `test` namespace on both `cluster1` and `cluster2`, pods in these namespaces can communicate freely across clusters. This show that, generally, network policies treat all clusters as a single unit, though they remain cluster-aware.

Reviewer 2: ```suggestion
The default allow-from-same-namespace policy applies within an OpenShift namespace.
While namespaces are unique within a single cluster, this constraint does not extend to a meshed cluster.
Since the policy matches based on namespace name, it applies to any namespace with the same name across clusters. For example, if you have a `test` namespace on both `cluster1` and `cluster2`, pods in these namespaces can communicate freely across clusters.
This show that, generally, network policies treat all clusters as a single unit, though they remain cluster-aware.
```

For stricter isolation, you must specify the cluster using the `io.cilium.k8s.policy.cluster` label. 

The following policy https://docs.cilium.io/en/latest/network/clustermesh/policy/#allowing-specific-communication-between-clusters[from upstream] illustrates how to allow particular pods to communicate between two clusters. 

[source,yaml]
----
apiVersion: "cilium.io/v2"

> kind: CiliumNetworkPolicy

Reviewer: I personally would stay with a regular `NetworkPolicy`. It's not a feature that needs `CiliumNetworkPolicy` and they might disappear or have breaking changes at some point.

Reviewer 2: Especially as we figured out there are (currently) issues with `CiliumNetworkPolicy` and valid policies block traffic.

metadata:
  name: "allow-cross-cluster"
spec:
  description: "Allow x-wing in cluster1 to contact rebel-base in cluster2"
  endpointSelector:
    matchLabels:
      name: x-wing
      io.cilium.k8s.policy.cluster: cluster1

>   egress:

Reviewer: Because the default policies are all ingress policies we should stick with an ingress policy.
